planet/src/main/webapp/js/placesItinerary.js

```diff
@@ -236,3 +245,32 @@ function validatePlaceDuration(duration) {
     }
     return true;
 }
+
+export function renderPlaceButtons(placeId, enableAdd) {
+    if (placeId) {
+        togglePlaceButton(placeId, enableAdd);
+    } else {
+        const userId = firebase.auth().currentUser.uid;
```

Comment:
It's safer to check that the currentUser is different than null before trying to access the uid property.

Reply:
I can quickly add it in, but there won't be any place buttons if the user is not signed in, so it's almost impossible for this edge case to happen

Reply:
hmm not sure, if the user is signed in I can't think of any other scenarios that the userid would be null

Reply:
If in you're sure that the function is only called when the user is signed in, then it should be good as it is. I usually take precautions and verify that currentUser is not null to avoid errors popping up in the console.

Reply:
Okay! thanks!

Reply:
What about network failure?

Reply:
true, i'll add it in then! thank you!